### PR TITLE
feat(rust): Add strategy that does json schema validation

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: rust_snuba/src/strategies/validate_schema.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        SchemaValidator {
+            schema,
+            enforce_schema,
+        }
+    }
+}
+
+impl TaskRunner<KafkaPayload, KafkaPayload> for SchemaValidator {
+    fn get_task(&self, message: Message<KafkaPayload>) -> RunTaskFunc<KafkaPayload> {
+        let mut errorred = false;
```
Comment (dbanda) [inline rust_snuba/src/strategies/validate_schema.rs:40]:
```suggestion
        let mut errored = false;
```

File: rust_snuba/src/strategies/validate_schema.rs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+        if let Some(schema) = &self.schema {
+            let payload = message.payload().payload.unwrap();
+
+            let res = schema.validate_json(&payload);
+
+            if let Err(err) = res {
+                log::error!("Validation error {}", err);
+                if self.enforce_schema {
+                    errorred = true;
```
Comment (dbanda) [inline rust_snuba/src/strategies/validate_schema.rs:50]:
```suggestion
                    errored = true;
```

File: rust_snuba/src/strategies/validate_schema.rs [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                };
+            }
+        }
+
+        Box::pin(async move {
+            if errorred {
```
Comment (dbanda) [inline rust_snuba/src/strategies/validate_schema.rs:56]:
```suggestion
            if errored {
```

File: rust_snuba/src/strategies/validate_schema.rs [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    inner: Box<dyn ProcessingStrategy<KafkaPayload>>,
+}
+
+impl ValidateSchema {
+    #[allow(dead_code)]
+    pub fn new<N>(next_step: N, topic: String, enforce_schema: bool) -> Self
+    where
+        N: ProcessingStrategy<KafkaPayload> + 'static,
+    {
+        let concurrency = 5;
```
Comment (dbanda) [inline rust_snuba/src/strategies/validate_schema.rs:75]: Should this be hardcoded?

Comment (lynnagara) [reply rust_snuba/src/strategies/validate_schema.rs]: Probably not, updated to an arg

**Review by untitaker — APPROVED** [review]

**Pull request merged**